### PR TITLE
feat(query): QualifiedName support 'db.table.*' and 'table.*'

### LINKED ISSUE #8906: Feature: support `a.*` in SQL
**Summary**

In a dbt snapshot SQL  case , the `a.*` in it and databend has the error as follows:

```
Unsupported indirection type
```

cc @sundy-li

### PR DESCRIPTION
**Pull request by TCeason**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Support 

select table.* from table 

and

select database.table.* from table 

Closes #8906

### CODE REVIEW

**Comment by TCeason** [comment]
The next step will support [exclude | except](https://github.com/datafuselabs/databend/issues/8964), in order not to block the cloud, to submit to the pr

**Comment by BohuTANG** [comment]
Need a stateless test for this patch.

**Comment by TCeason** [comment]
> Need a stateless test for this patch.

Ohh sry, it's on my local... forget to commit.

File: src/query/sql/src/planner/binder/project.rs [hunk 1]
```diff
@@ -193,9 +173,94 @@ impl<'a> Binder {
...
+                                        match_table = true;
+                                        let select_item = SelectItem {
+                                            select_target,
+                                            scalar: BoundColumnRef {
+                                                column: column_binding.clone(),
+                                            }
+                                            .into(),
+                                            alias: column_binding.column_name.clone(),
+                                        };
+                                        println!("select_item is {:?}", select_item);
```
Comment (leiysky) [inline src/query/sql/src/planner/binder/project.rs:202]: Should remove this?

Comment (TCeason) [reply src/query/sql/src/planner/binder/project.rs]: Done

**Review by leiysky — APPROVED** [review]

**Comment by BohuTANG** [comment]
cc @hantmac

**Pull request merged**